src/main/scala/nl.knaw.dans.easy.bagstore/package.scala

```diff
@@ -137,4 +139,11 @@ package object bagstore {
     }
     list.toSet
   }
+
+  def getUUID(uuidStr: String): UUID = {
+    uuidStr.toUUID.toTry match {
+      case Success(uuid) => uuid
+      case Failure(e) => throw new IllegalArgumentException(e.getMessage)
+    }
+  }
```

Comment:
I see why you did it this way. It requires the least code changes and it's consistent in types with the old code. Of course I want to point out that the type here is not honest about what it returns: it will throw an error on invalid input. In most cases this works out fine as the `getUUID` is wrapped in a `Try`. There are however some places where this is not the case and I would like you to at least fix those. I'll point them out in comments below.

Also I would like to see a `@throws[IllegalArgumentException]("some message with a reason for throwing this error")` right above the function to document the exception throwing.

Finally, have you considered keeping the `UUIDError` as is, rather than converting it to an `IllegalArgumentException`? In that case you should also check whether the `IllegalArgumentException` is still necessary in the pattern match in the error handling of the servlet(s).

Reply:
I added `@throws[IllegalArgumentException]("String '$uuidStr' is not a UUID")` above the `getUUID` method.
I rather give in that method `IllegalArgumentException` because `IllegalArgumentException` is caught in so many places in the code.